File: Data Structures & Algorithms/is-palindrome/submission-2.java
class Solution {
    public boolean isPalindrome(String s) {
        // initialize left and right pointers
        int left = 0;
        int right = s.length()-1;

        // while: left < right
        //// get next alphanumeric left – while: left< right char indexed left isn't alpha., increment
        //// get next alphanumeric right – while: left < r and char indexed right isn't alpha., decrement
        //// if: left char != right char, return false 
        //// else: update left and right pointer
        while (left < right) {
            while (left < right && !Character.isLetterOrDigit(s.charAt(left)))
                left++;
            while (right > left && !Character.isLetterOrDigit(s.charAt(right)))
                right--;
            
            if (Character.toLowerCase(s.charAt(left)) != Character.toLowerCase(s.charAt(right)))
                return false;

            left++;
            right--;
        }

        // exits loop: return true
        return true;
        
    }
}

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        # initialize maxProfit to zero – assumes no transaction
        # initialize left=0 and right=1 pointers
        maxProfit = 0
        left = 0
        right = 1

        # while: right is not out of bound – right < list.length
        ## calculate profit
        ## if: profit < 0, update pointers: left = right, right = left+1
        ## else: update maxProfit and increment right pointer
        while right < len(prices):
            profit = prices[right] - prices[left]
            if profit < 0:
                left = right
                right = left+1
            else:
                maxProfit = max(maxProfit, profit)
                right += 1

        # return maxProfit
        return maxProfit